### ISSUE TITLE
Error: Missing IncomingMessage property on http2 module?

### ISSUE THREAD

**Issue by talwaserman** [Unchosen/express-http2-workaround#3]
getting the following error 
```Error: Missing IncomingMessage property on http2 module?```
npm verison: 6.4.1
node version: 11.1.0
express: 4.15.4

**Comment by SangeetaKhan** [comment]
Yeah even I am getting the same error. Tried working with the requestHTTP2 and responseHTTP2 methods, but failed. Needs help.

<img width="1208" alt="screen shot 2018-11-06 at 1 44 55 pm" src="https://user-images.githubusercontent.com/8016610/48056953-e2db2b80-e1d8-11e8-9f27-2830f0f41d87.png">

**Comment by Abourass** [comment]
Same.

**Comment by Jashepp** [comment]
NodeJS version 11.0.0+ has a [core http2 module](https://nodejs.org/dist/latest-v11.x/docs/api/http2.html) (previously hidden behind `--expose-http2` command-line option).
Using `require('http2')` will use that core module _instead_ of [molnarg's node-http2 module](https://www.npmjs.com/package/http2) (which is now deprecated).

You may need to require the old http2 module via a relative or fixed path instead then pass it to express-http2-workaround.
I need to update this to check if the core module is passed in or not, to have a more relevant error message be thrown. I'll update the README to mention this information too when I can.

**Comment by talwaserman** [comment]
can you please provide code snippet of the solution?
is there also a way to use this workaround with the new http2 module ?

**Comment by SangeetaKhan** [comment]
Any update on this ?

**Comment by SangeetaKhan** [comment]
@Jashepp Please share me any code for the same

**Comment by niklasHagner** [comment]
Stumbled upon this thread as I googled for this specific error.
I had an old project that hadn't been updated since 2017 and while running it with node 10 I got the error. (It did work fine on older versions like version 7 though)

As Jasheep said the issue is that node (since version 10 LTS) now has a native `http2` so `require("http2")` will load that one. instead of  https://www.npmjs.com/package/http2

The solution is to replace `require("http2");` with `require("./node_modules/http2");` and all the code examples from the readme will work fine.

@Jashepp - I know this project hasn't been updated in a long time but a suggestion be to update the code in the README with this small change.



-----------

So @SangeetaKhan , @talwaserman - the code sample from https://github.com/Jashepp/express-http2-workaround#example works if you just change the path for the http2 import.

Here's a full example I used, works fine in node 10.16.3.
```
var express = require('express');
var http2 = require("./node_modules/http2"); //here's the significant change! this loads the module from https://www.npmjs.com/package/http2 instead of node's built-in
var fs = require("fs");
var app = express();

// Make HTTP2 work with Express (this must be before any other middleware)
require('express-http2-workaround')({ express:express, http2:http2, app:app });

// Setup HTTP/2 Server
var httpsOptions = {
    'key' : fs.readFileSync(__dirname + '/keys/ssl.key'),
    'cert' : fs.readFileSync(__dirname + '/keys/ssl.crt'),
    'ca' : fs.readFileSync(__dirname + '/keys/ssl.crt')
};
var http2Server = http2.createServer(httpsOptions,app);
var port = 5000;
http2Server.listen(port,function(){
  console.log("Express HTTP/2 server started on", port);
});

// Serve some content
app.get('/', function(req,res){
    res.send('Hello World!');
});
```

Browse to `https://localhost:5000` to view the output.

PS: the original example uses 443 but some machines have permission denied for those ports so I changed it to a random one like 5000.
DS: make sure you use https and not http as this example only sets up the https-variant.
